frontend/public/components/create-yaml.tsx

```diff
@@ -27,7 +31,7 @@ export const CreateYAML = connectToPlural((props: CreateYAMLProps) => {
   } = props;
   const { params } = match;
   const { t } = useTranslation();
-  const templateExtensions = useExtensions<YAMLTemplate>(isYAMLTemplate);
+  const [templateExtensions] = useResolvedExtensions<YAMLTemplate>(isYAMLTemplate);
```

Comment:
You only need to resolve the templates which will actually be used. That could be done by augmenting the guard.
```suggestion
  const [templateExtensions] = useResolvedExtensions<YAMLTemplate>((e) => isYAMLTemplate(e) && ...);
```

You only want to resolve code refs for the extensions that are actually being used.

Resolution of templates is now asynchronous. To prevent flashing, you should handle the loading case and show the `LoadingBox` otherwise there's potential to show a different template before the extensions load.